Client/Frontend/Brave Today/Composer/FeedDataSource.swift

```diff
@@ -345,49 +340,67 @@ class FeedDataSource {
     var items: [FeedItem.Content]
   }
 
-  private func loadRSSLocation(_ location: RSSFeedLocation) -> Deferred<Result<RSSDataFeed, Error>> {
-    let deferred = Deferred<Result<RSSDataFeed, Error>>(value: nil, defaultQueue: .main)
+  private func loadRSSLocation(_ location: RSSFeedLocation) async throws -> RSSDataFeed {
     let parser = FeedParser(URL: location.url)
-    parser.parseAsync { [weak self] result in
-      switch result {
-      case .success(let feed):
-        if let source = FeedItem.Source(from: feed, location: location) {
-          var content: [FeedItem.Content] = []
-          switch feed {
-          case .atom(let atomFeed):
-            if let feedItems = atomFeed.entries?.compactMap({ entry -> FeedItem.Content? in
-              FeedItem.Content(from: entry, location: location)
-            }) {
-              content = feedItems
-            }
-          case .rss(let rssFeed):
-            if let feedItems = rssFeed.items?.compactMap({ entry -> FeedItem.Content? in
-              FeedItem.Content(from: entry, location: location)
-            }) {
-              content = feedItems
-            }
-          case .json(let jsonFeed):
-            if let feedItems = jsonFeed.items?.compactMap({ entry -> FeedItem.Content? in
-              FeedItem.Content(from: entry, location: location)
-            }) {
-              content = feedItems
+    return try await withCheckedThrowingContinuation { continuation in
+      parser.parseAsync { [weak self] result in
+        switch result {
+        case .success(let feed):
+          if let source = FeedItem.Source(from: feed, location: location) {
+            var content: [FeedItem.Content] = []
+            switch feed {
+            case .atom(let atomFeed):
+              if let feedItems = atomFeed.entries?.compactMap({ entry -> FeedItem.Content? in
+                FeedItem.Content(from: entry, location: location)
+              }) {
+                content = feedItems
+              }
+            case .rss(let rssFeed):
+              if let feedItems = rssFeed.items?.compactMap({ entry -> FeedItem.Content? in
+                FeedItem.Content(from: entry, location: location)
+              }) {
+                content = feedItems
+              }
+            case .json(let jsonFeed):
+              if let feedItems = jsonFeed.items?.compactMap({ entry -> FeedItem.Content? in
+                FeedItem.Content(from: entry, location: location)
+              }) {
+                content = feedItems
+              }
             }
+            guard let self = self else { return }
```

Comment:
no need to put continuation in this block as well?

Reply:
Will add a resume here for safety